### PR TITLE
Fix inconsistencies and typos in argument names

### PR DESCRIPTION
**Pull request by bcr3ative**
This PR should fix issue #15668 .

**Note**: I mostly changed argument names in header files to match those found in their definition (.cpp), but in some cases I changed them in their definition as well to keep naming consistent across respective files.

I am however not sure how to approach the changes in `gdnative_library_editor_plugin.h` . I changed `*item` to `*p_item` to mirror the naming found in the definition file but this breaks consistency because argument names in the header file (in other functions) do not follow the `p_[argument_name]` naming convention. I could change the argument name to `item` in definition but there is a local scope variable with the same name in the function.

### CODE REVIEW

File: modules/bullet/space_bullet.h [hunk 1]
```diff
@@ -75,7 +75,7 @@ class BulletPhysicsDirectSpaceState : public PhysicsDirectSpaceState {
 
 	virtual int intersect_point(const Vector3 &p_point, ShapeResult *r_results, int p_result_max, const Set<RID> &p_exclude = Set<RID>(), uint32_t p_collision_mask = 0xFFFFFFFF);
 	virtual bool intersect_ray(const Vector3 &p_from, const Vector3 &p_to, RayResult &r_result, const Set<RID> &p_exclude = Set<RID>(), uint32_t p_collision_mask = 0xFFFFFFFF, bool p_pick_ray = false);
-	virtual int intersect_shape(const RID &p_shape, const Transform &p_xform, float p_margin, ShapeResult *r_results, int p_result_max, const Set<RID> &p_exclude = Set<RID>(), uint32_t p_collision_mask = 0xFFFFFFFF);
+	virtual int intersect_shape(const RID &p_shape, const Transform &p_xform, float p_margin, ShapeResult *p_results, int p_result_max, const Set<RID> &p_exclude = Set<RID>(), uint32_t p_collision_mask = 0xFFFFFFFF);
```
Comment (akien-mga) [inline modules/bullet/space_bullet.h]: This one should actually be renamed to `r_results` in the .cpp, the `r_` prefix is for parameters that are modified by the method.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
> I am however not sure how to approach the changes in `gdnative_library_editor_plugin.h` . I changed `*item` to `*p_item` to mirror the naming found in the definition file but this breaks consistency because argument names in the header file (in other functions) do not follow the `p_[argument_name]` naming convention. I could change the argument name to `item` in definition but there is a local scope variable with the same name in the function.

CC @Geequlim @karroffel

File: modules/mono/csharp_script.h [hunk 2]
```diff
@@ -172,7 +172,7 @@ class CSharpInstance : public ScriptInstance {
 	bool base_ref;
 	bool ref_dying;
 
-	void _ml_call_reversed(MonoObject *p_mono_object, GDMonoClass *klass, const StringName &p_method, const Variant **p_args, int p_argcount);
+	void _ml_call_reversed(MonoObject *p_mono_object, GDMonoClass *p_klass, const StringName &p_method, const Variant **p_args, int p_argcount);
```
Comment (vnen) [inline modules/mono/csharp_script.h]: This could be `p_class`, since that is not a reserved word.

Comment (neikeq) [reply modules/mono/csharp_script.h]: This method is being removed by #16205.

**Comment by hpvb** [comment]
If this changes GDNative bindings in any way, or will require recompilation of existing GDNative modules built against 3.0.0 I'd much rather not pull this into 3.0.

I'm totally in favor of this for master/3.1 though.

**Comment by akien-mga** [comment]
It doesn't modify any bindings, so that should be fine to cherry-pick.

**Comment by akien-mga** [comment]
@bcr3ative Could you rebase and amend the comment to do the suggested changes? See [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase) for rebasing howto.

**Comment by bcr3ative** [comment]
Rebase and amend done.

**Pull request merged**